site/src/components/Resources/PortForwardButton.tsx

```diff
@@ -0,0 +1,274 @@
+            ? "No open ports were detected."
+            : "Port URLs are only accessible by you."}
+        </HelpTooltipText>
+        <Box sx={{ marginTop: (theme) => theme.spacing(1.5) }}>
+          {ports?.map((p) => {
+            const url = portForwardURL(
+              host,
+              p.port,
+              agentName,
+              workspaceName,
+              username,
+            )
+            const label =
+              p.process_name !== "" ? p.process_name : `Port ${p.port}`
+            return (
+              <Link
+                underline="none"
+                sx={{
+                  color: (theme) => theme.palette.text.primary,
+                  fontSize: 14,
+                  display: "flex",
+                  alignItems: "center",
+                  gap: 1,
+                  py: 0.5,
+                  fontWeight: 500,
+                }}
+                key={p.port}
+                href={url}
+                target="_blank"
+                rel="noreferrer"
+              >
+                <OpenInNewOutlined sx={{ width: 14, height: 14 }} />
+                {label}
+                <Box
+                  component="span"
+                  sx={{
+                    ml: "auto",
+                    color: (theme) => theme.palette.text.secondary,
+                    fontSize: 13,
+                    fontWeight: 400,
+                  }}
+                >
+                  {p.port}
+                </Box>
+              </Link>
+            )
+          })}
+        </Box>
+      </Box>
+
+      <Box
+        sx={{
+          padding: (theme) => theme.spacing(2.5),
+        }}
+      >
+        <HelpTooltipTitle>Open app</HelpTooltipTitle>
+        <HelpTooltipText
+          sx={{ color: (theme) => theme.palette.text.secondary }}
+        >
+          Access ports running on the agent with the port, agent name, workspace
+          name and your username.
+        </HelpTooltipText>
+
+        <Box
+          component="form"
+          sx={{
+            border: (theme) => `1px solid ${theme.palette.divider}`,
+            borderRadius: "4px",
+            mt: 2,
+            display: "flex",
+            alignItems: "center",
+            "&:focus-within": {
+              borderColor: (theme) => theme.palette.primary.main,
+            },
+          }}
+          onSubmit={(e) => {
+            e.preventDefault()
+            const formData = new FormData(e.currentTarget)
+            const port = Number(formData.get("portNumber"))
+            const url = portForwardURL(
+              host,
+              port,
+              agentName,
+              workspaceName,
+              username,
+            )
+            window.open(url, "_blank")
+          }}
+        >
+          <Box
+            aria-label="Port number"
+            name="portNumber"
+            component="input"
+            type="number"
+            placeholder="Type a port number..."
+            min={0}
```

Comment:
min: 0
max: 65535

Actually, I'm not sure about 0.. as there are some Coder [internal ports](https://github.com/coder/coder/blob/main/tailnet/conn.go#L48-L52). Maybe you should start with 10?